sympy/physics/units/util.py

```diff
@@ -154,6 +160,46 @@ def quantity_simplify(expr):
         ref = v[0]/v[0].scale_factor
         expr = expr.xreplace({vi: ref*vi.scale_factor for vi in v[1:]})
 
+    if across_dimensions:
+        # combine quantities of different dimensions into a single
+        # quantity that is equivalent to the original expression
+
+        unit_system = UnitSystem.get_unit_system(unit_system)
+        dimension_system: DimensionSystem = unit_system.get_dimension_system()
+        dim_expr = unit_system.get_dimensional_expr(expr)
+        dim_deps = dimension_system.get_dimensional_dependencies(dim_expr, mark_dimensionless=True)
+
+        target_dimension = None
+        for result_dim, result_deps in dimension_system.dimensional_dependencies.items():
+            if result_deps == dim_deps:
+                target_dimension = result_dim
+                break
+
+        from . import volt, ohm, ampere, pascal, farad, second, watt, henry, meter, newton, weber, siemens, coulomb, tesla, candela, steradian
+        from . import voltage, impedance, current, pressure, capacitance, time, power, inductance, volume, force, luminous_intensity, magnetic_flux, conductance, charge, magnetic_density
+
+        preferred_dimension_to_unit = {
```

Comment:
This dictionary is assuming the base units of _SI_. I think these values can be extracted from the `unit_system` variable. This would make the code independent of the unit system.

Reply:
I agree. I'm unsure if this is possible without adding additional metadata to `Quantity`. For example, `bar` and `pascal` are both units of pressure. How do we know which one to pick?

I'll try it out and see what happens.

Reply:
The unit system object should have a dictionary of scale factors, IIRC. Maybe that could be used to determine the base unit?

Reply:
I tried filtering units by scale factors, and I was getting some weird results. I'll push a new commit to show my current solution momentarily.